lib/ex_aws/rekognition.ex

```diff
@@ -681,9 +455,24 @@ defmodule ExAws.Rekognition do
     S3Object.map(object)
   end
 
-  defp request(action, data) do
-    http_method = @actions |> Map.fetch!(action)
+  defp stringify_enum_opts(opts, keys) do
+    {enum_opts, opts} = Keyword.split(opts, keys)
+    opts ++ Enum.map(enum_opts, fn {k, v} -> {k, stringify_enum(v)} end)
+  end
 
+  defp stringify_enum(value) do
+    value |> Atom.to_string() |> String.upcase()
+  end
+
+  defp map_notification_channel(opts) do
+    if value = opts[:notification_channel] do
+      Keyword.replace!(opts, :notification_channel, NotificationChannelObject.map(value))
+    else
+      opts
+    end
+  end
```

Comment:
Not sure if I would use pattern matching instead of an if, what do you think?

Reply:
I'm sorry for the latency. To be honest I wasn't sure how I'd employ pattern matching in this case ("does the kw list contain a pair with key X?") and went with the if-expression instead. I'd be more than happy to implement it in some more functional / Elixiry fashion, and will have a look at this again.

Reply:
@everila can you leave it as it was before and ignore what I said? I will try to merge it into what we have 😄

Reply:
Thanks 👍